package BLUE_BERRY.Test;

// sample type
enum TestType {
	TYPE_1 = 0;								// 모든 서버를 대표
	TYPE_2 = 1;
	TYPE_3 = 2;
}

// Error type
enum ErrorCode {
	ERR_SUCCESS = 0;
	ERR_LOGIN_FAIL = 1;
	ERR_ARGUMENT_FAIL = 2;
}


// sample struct
message TestMessage {

	enum SmallType {
		NORMAL = 0;
		HARD = 1;
		HELL = 2;
	}
	
	required string testString = 1;
	required uint32 testUint32 = 2;
	required SmallType testSmallType = 3;	
	optional TestType testTestType = 4 [default = TYPE_1];
	required bool testBool = 5;
	required int32 testInt32 = 6;
	optional uint64 testUint64 = 7;
	optional float testFloat = 8;
}


// sample mseeage
message LoginReq {
	required string id = 1;
}

message LoginAns {
	required ErrorCode err = 1;	
}